go/pkg/tree/tree.go

```diff
@@ -138,9 +136,6 @@ func buildTree(files map[string]*chunker.Chunker) *treeNode {
 }
 
 func packageTree(t *treeNode, chunkSize int) (root digest.Digest, blobs map[digest.Digest]*chunker.Chunker, err error) {
-	if t == nil {
-		return digest.Empty, nil, errors.New("nil treeNode while packaging tree")
-	}
```

Comment:
Why remove this?

Reply:
Because it can't happen. Actually, I want to remove a lot of other checks as well. Public functions should error-check for conditions that are parameter or system triggered, and these error checks should be covered  by tests. Checking for invariants that your own private code creates is, imo redundant and should be avoided. Returning an error is part of the public contract. If we violate our own private invariants, we don't need to return an error, we need to crash.
So I want to remove all these checks.

Reply:
I'm not sure what that gets us. This is either covered by tests or not. Panicing doesn't add any extra level of visibility. And the function already returns an error, so I feel like with this rule we have to assess whether the error condition is an internal invariant or not (for instance, is proto.Marshal failing expected or not? In general, "Don't panic" is preferable unless there is absolutely no way of recovery.

Reply:
Discussed offline, removed more checks.